python/cuml/test/test_random_forest.py

```diff
@@ -656,21 +637,27 @@ def test_rf_memory_leakage(fil_sparse_format, column_info, nrows):
     # before the first call to get_memory_info.
     base_model = curfc(handle=handle)
     base_model.fit(X_train, y_train)
+    handle.sync()  # just to be sure
     free_mem = cuda.current_context().get_memory_info()[0]
 
-    rfc_model = curfc(handle=handle)
-    rfc_model.fit(X_train, y_train)
+    def test_for_memory_leak():
+        cuml_mods = curfc(handle=handle)
+        cuml_mods.fit(X_train, y_train)
+        handle.sync()  # just to be sure
+        # Calculate the memory free after fitting the cuML model
+        delta_mem = free_mem - cuda.current_context().get_memory_info()[0]
+        assert delta_mem == 0
 
-    # Calculate the memory free after fitting the cuML RF model
-    delta_mem = free_mem - cuda.current_context().get_memory_info()[0]
-    cuml_mods = curfc(handle=handle)
-    cuml_mods.fit(X_train, y_train)
+        for i in range(3):
+            cuml_mods.predict(X_test, predict_model="GPU",
+                              fil_sparse_format=fil_sparse_format)
+            handle.sync()  # just to be sure
+            # Calculate the memory free after predicting the cuML model
+            delta_mem = free_mem - cuda.current_context().get_memory_info()[0]
+            assert delta_mem == 0
 
     for i in range(n_iter):
-        cuml_mods.predict(X_train, predict_model="GPU")
-        handle.sync()
-        delta_mem = free_mem - cuda.current_context().get_memory_info()[0]
-        assert delta_mem == 0
+        test_for_memory_leak()
 
 
 @pytest.mark.parametrize('max_features', [1.0, 'auto', 'log2', 'sqrt'])
```

Comment:
for predict_proba test we can remove the `split_algo` variable so that the test does not run for two different values of it.